numpy/random/mtrand/generate_randint_helpers.py

```diff
@@ -0,0 +1,152 @@
+"""
+Helper file for generating the necessary helper functions
+for each `dtype` in `np.random.randint`. The function
+`gen_randint_helpers` is called right before Cythonization
+in NumPy setup.
+
+"""
+
+# npy_dt : Desired C API data type
+# npy_udt : Unsigned version of desired C
+#           API data type (could be identical)
+# npy_dt : Desired Python API data type
+
+template = """
+def _rand_{npy_dt}(low, high, size, rngstate):
```

Comment:
The cost of the `if-else` is that you have to add the extra code to wrap structs as python objects and have uncallable functions that someone could try and call.  SO there are trade-offs in either direction.